src/core/control/ToolHandler.cpp

```diff
@@ -577,6 +577,18 @@ auto ToolHandler::isSinglePageTool() const -> bool {
            toolType == TOOL_SELECT_PDF_TEXT_RECT;
 }
 
+auto ToolHandler::supportsTapFilter() const -> bool {
+    ToolType toolType = this->getToolType();
+
+    return toolType == TOOL_PEN || toolType == TOOL_HIGHLIGHTER || toolType == TOOL_DRAW_RECT ||
+           toolType == TOOL_DRAW_ELLIPSE || toolType == TOOL_DRAW_COORDINATE_SYSTEM || toolType == TOOL_DRAW_ARROW ||
+           toolType == TOOL_DRAW_DOUBLE_ARROW || toolType == TOOL_DRAW_SPLINE;
+    // Add toolType == TOOL_ERASER
+    // Add? || toolType == TOOL_SELECT_REGION || toolType == TOOL_SELECT_RECT || toolType == TOOL_SELECT_OBJECT  ||
+    // toolType == TOOL_FLOATING_TOOLBOX || toolType == TOOL_SELECT_PDF_TEXT_LINEAR || toolType ==
+    // TOOL_SELECT_PDF_TEXT_RECT
+}
```

Comment:
Here is something that should be discussed, I think.

Previously, the tap filter only existed for a handful of tools. If another tool was selected, tapping would not show the floating toolbox.
With this PR, it becomes easy to add more tools to the "tap detecting" list.
Do we want to? If so, which ones?

The question is: do we want to allow more tools?

Reply:
I think we should. At very least we should add the eraser. I can't see any downside of it. There are no eraser tapping actions, so no overlap. See also #1863 where this has been requested.

In general I would let any tool support the tap filter that doesn't need tap actions by itself. For the selection tools (which also include multilayer rectangle and region tools by now) tap allows to select an object (and maybe on a different layer). For the pdf selection tools tap may allow to follow a link. So that has to be checked carefully.

One thing that is annoying with the floating toolbox is the fact, that once you switch (via the floating toolbox) to a tool that does not support the tap filter, you can't get the floating toolbox up that easily. You can't even put it onto a toolbar (even though that could be changed). That would also be an argument for letting more tools support the tap filter.

Reply:
Experimenting with selection tools and PDF selection tools added to the tap filter, I would advise against adding them. For PDF selection tools it is annoying that you can't just click on a link any more to get the popup displayed. 
For selection tools it currently doesn't work at all (the selection action is still started and you get a crash).
Maybe it's wiser to just add the floating toolbox as a toolbar item via 
```c++    
addCustomItem("FLOATING_TOOLBOX", ACTION_TOOL_FLOATING_TOOLBOX, "floating-toolbox", _("Floating Toolobx"));
```
and by adding `FLOATING_TOOLBOX` to the `Floating toolbox (experimental)]` toolbar in `toolbar.ini` with according changes to `TOOLBAR_INI_HEADER` in `ToolbarModel.cpp`.

Reply:
I guess all tools should support the floating toolbox tap thingy, but only if the tool does not care about the tap: so, if the tap is not on a link, the PDF selection tools should open the floating toolbox; if there is nothing to select at the tap location, the section tools should open the floating toolbox.

It should not be too hard to do, it's just about prioritizing event responses by adding a couple of tool-specific lines in `PageView::onTapEvent()`. It probably goes a bit beyond this PR though.